OpenRA.Mods.Common/Activities/Air/FlyAttack.cs

```diff
@@ -28,9 +28,18 @@ public override Activity Tick(Actor self)
 			if (!target.IsValidFor(self))
 				return NextActivity;
 
-			var limitedAmmo = self.TraitOrDefault<LimitedAmmo>();
-			if (limitedAmmo != null && !limitedAmmo.HasAmmo())
-				return NextActivity;
+			var limitedAmmo = self.TraitsImplementing<LimitedAmmo>();
+			if (limitedAmmo != null)
+			{
+				foreach (var la in self.TraitsImplementing<LimitedAmmo>())
+				{
+					// Skip LimitedAmmo pools that SelfReload or still have ammo)
+					if (la.ReloadsAutomatically() || la.HasAmmo())
+						continue;
+
+					return NextActivity;
+				}
+			}
```

Comment:
Some people argue that every `if` or loop statement should have brackets. In OpenRA we use them only when we have to. And if you have more than one line, you **need** to group the lines using brackets. (the `foreach` here and it's block are considered one line)